pyfr/backends/openmp/types.py

```diff
@@ -182,7 +182,23 @@ def group(self, kerns, subs=[]):
             groups.append(rargs)
 
         # Arrange for the groupings to be inserted into the final run list
-        self.kins[max(self.knodes[k] for k in kerns) - 1] = groups
+        gdeps = []
+        for k in kerns:
+            gdeps = gdeps + [dep for dep in self.kdeps[k] if dep not in kerns]
+        gid = None
+        if gdeps:
+            lk = max(self.knodes[dep] for dep in gdeps)
+            gid = min(self.knodes[k] for k in kerns if self.knodes[k] > lk) - 1
+        else:
+            gid = min(self.knodes[k] for k in kerns) - 1
+        self.kins[gid] = groups
```

Comment:
Add newline.

Reply:
Avoid `\`, they should almost never be used.  Newlines are allowed inside [] and ().

Reply:
Also, we might want to move down the `self.kins[gid] = groups` bit until after the sanity check.

Reply:
I think I've now fixed this